java-lib/src/main/java/com/yammer/metrics/core/DeltaCounter.java

```diff
@@ -0,0 +1,40 @@
+package com.yammer.metrics.core;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.wavefront.common.MetricConstants;
+
+/**
+ * A counter for Wavefront delta metrics.
+ *
+ * Differs from a counter in that it is reset in the SocketMetricsProcessor every time the value is
+ * reported.
+ *
+ * (This is similar to how WavefrontHistogram is implemented)
+ *
+ * @author Suranjan Pramanik (spramanik@wavefront.com)
+ */
+public class DeltaCounter extends Counter {
+
+    private DeltaCounter() {
+        // nothing to do. keeping it private so that it can't be instantiated directly
+    }
+
+    @VisibleForTesting
+    public static synchronized DeltaCounter get(MetricsRegistry registry, MetricName metricName) {
+        if (registry == null || metricName == null || metricName.getName().isEmpty()) {
+            throw new IllegalArgumentException("Invalid arguments");
+        }
+
+        String name = metricName.getName();
+        if (!(name.startsWith(MetricConstants.DELTA_PREFIX) || name.startsWith(MetricConstants
```

Comment:
There are bunch of places where we have 
name.startsWith(MetricConstants.DELTA_PREFIX) || name.startsWith(MetricConstants
                .DELTA_PREFIX_2)
might want to put that in a static method 

boolean isDelta(String name) {

}

Reply:
Added a static method.